alicloud/service_alicloud_cen.go

```diff
@@ -58,3 +64,94 @@ func (client *AliyunClient) WaitForCenInstance(cenId string, status Status, time
 
 	return nil
 }
+
+func (client *AliyunClient) DescribeCenAttachedChildInstanceById(instanceId, cenId string) (c cbn.ChildInstance, err error) {
+	request := cbn.CreateDescribeCenAttachedChildInstancesRequest()
+	request.CenId = cenId
+
+	for pageNum := 1; ; pageNum++ {
+		request.PageNumber = requests.NewInteger(pageNum)
+		response, err := client.cenconn.DescribeCenAttachedChildInstances(request)
+		if err != nil {
+			return c, err
+		}
+
+		instanceList := response.ChildInstances.ChildInstance
+		for instanceNum := 0; instanceNum <= len(instanceList)-1; instanceNum++ {
+			if instanceList[instanceNum].ChildInstanceId == instanceId {
+				return instanceList[instanceNum], nil
+			}
+		}
+
+		if pageNum*response.PageSize >= response.TotalCount {
+			return c, GetNotFoundErrorFromString(GetNotFoundMessage("CEN Child Instance", instanceId))
+		}
+	}
+}
+
+func (client *AliyunClient) WaitForCenChildInstanceAttached(instanceId string, cenId string, status Status, timeout int) error {
+	if timeout <= 0 {
+		timeout = DefaultTimeout
+	}
+
+	for {
+		instance, err := client.DescribeCenAttachedChildInstanceById(instanceId, cenId)
+		if err != nil {
+			return err
+		}
+		if instance.Status == string(status) {
+			break
+		}
+		timeout = timeout - DefaultIntervalShort
+		if timeout <= 0 {
+			return GetTimeErrorFromString(GetTimeoutMessage("CEN Child Instance Attachment", string(status)))
+		}
+		time.Sleep(DefaultIntervalShort * time.Second)
+	}
+
+	return nil
+}
+
+func (client *AliyunClient) WaitForCenChildInstanceDetached(instanceId string, cenId string, timeout int) error {
+	if timeout <= 0 {
+		timeout = DefaultTimeout
+	}
+
+	for {
+		_, err := client.DescribeCenAttachedChildInstanceById(instanceId, cenId)
+		if err != nil {
+			if NotFoundError(err) {
+				break
+			} else {
+				return err
+			}
+		}
+
+		timeout = timeout - DefaultIntervalShort
+		if timeout <= 0 {
+			return GetTimeErrorFromString(fmt.Sprintf("Waitting for %s detach timeout.", instanceId))
+		}
+		time.Sleep(DefaultIntervalShort * time.Second)
+	}
+
+	return nil
+}
+
+func getCenIdAndAnotherId(id string) (string, string, error) {
+	parts := strings.Split(id, ":")
+
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("invalid resource id")
+	}
+	return parts[0], parts[1], nil
+}
+
+func getCenInstanceType(id string) (c string, e error) {
+	if strings.HasPrefix(id, "vpc") {
+		return "VPC", nil
+	} else if strings.HasPrefix(id, "vbr") {
+		return "VBR", nil
+	} else {
+		return c, fmt.Errorf("CEN child instance ID invalid, should start with \"vpc\" or \"vbr\"")
```

Comment:
Please modify line 155:  `return c, fmt.Errorf("CEN child instance ID invalid. Now, it only supports VPC or VBR" instance.)`